webpack.config.js

```diff
@@ -1,17 +1,28 @@
+require('dotenv').config();
 const path = require('path');
+const glob = require('glob-all');
 const webpack = require('webpack');
+const TerserPlugin = require('terser-webpack-plugin');
+const PurgecssPlugin = require('purgecss-webpack-plugin');
 const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const OptimizeCSSAssetsPlugin = require('optimize-css-assets-webpack-plugin');
+const root = __dirname.substring(0, __dirname.lastIndexOf('/') + 1);
```

Comment:
Please enlighten me on this root.

Reply:
what root does is to get the path to the root of the application... the folder where server, build, client, package.json are.... and then appends the public string to the path.. ( its meant to always work from a file within the first level of build, server or any sibling folder to public )... it gives this whether HOST_URL/public .......

Reply:
i used the path resolve in the end